File: 26_RemoveDuplicatesfromSortedArray/removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 快慢指针解法，pre指向当前不重复元素
        pre = 0
        for idx, num in enumerate(nums):
            if idx > pre and nums[idx] != nums[pre]:
                pre += 1
                nums[pre] = nums[idx]
        return pre + 1

File: 26_RemoveDuplicatesfromSortedArray/removeDuplicates.go
func removeDuplicates(nums []int) int {
    // 快慢指针解法, pre指向当前不重复的元素的索引
    pre := 0
    for idx, _ := range nums {
        if pre < idx && nums[pre] != nums[idx] {
            pre += 1
            nums[pre] = nums[idx]
        }
    }

    return pre + 1
}
